core/src/main/java/org/infinispan/marshall/core/ExternallyMarshallable.java

```diff
@@ -2,18 +2,18 @@
 
 import java.io.Serializable;
 
-// Temporary class, should become active by default when testing
 final class ExternallyMarshallable {
 
    private ExternallyMarshallable() {
       // Static class
    }
 
-   public static boolean isAllowed(Object obj) {
+   static boolean isAllowed(Object obj) {
       String pkg = obj.getClass().getPackage().getName();
       boolean isBlackList =
             obj instanceof Serializable
             && isMarshallablePackage(pkg)
+            && !(obj instanceof ExternalPojo)
             && !isWhiteList(obj.getClass().getName());
```

Comment:
I'm not sure how this is supposed to work for user classes, as users can't modify the whitelist and they can't always modify their classes to extend `ExternalPojo` either. They could write+configure externalizers, but do we want to force them to write externalizers for all their classes?

My thinking was that we'd only have the blacklist active during the tests, so we'd only need `ExternalPojo` for our internal classes.